_posts/2021-05-04-how-to-update-your-ref-arch.adoc

```diff
@@ -0,0 +1,485 @@
+---
+title: How to update your v1 Reference Architecture to Service Catalog
```

Comment:
The reason it's not focusing on Ref Arch 2.0 is because we kicked out Gruntwork Pipelines and LZ migration as out of scope. Gruntwork Pipelines ends up being a deploy from scratch situation (which is covered by https://gruntwork.io/guides/automations/how-to-configure-a-production-grade-ci-cd-setup-for-apps-and-infrastructure-code/), and the amount of effort involved in resolving all the imports to make LZ work felt like it was too large in scope that it will significantly extend the timeline.

Note that this is largely driven by the issue that opening up service catalog to all customers is blocked on this guide, and the longer we go with the service catalog closed, the longer we have to maintain 3 copies of landing zone (`terraform-aws-security`, `terraform-aws-service-catalog`, and `terraform-aws-cis-service-catalog`).

Given that, this guide is really only focused on using Service Catalog + `dependency` blocks.

Reply:
Ok despite what I said, here is what I propose for handling this:

- Make the update to scope this as a v1 => v2 upgrade and describe the differences as you mentioned.
- Add two help boxes: one indicating that the landing zone migration guide will be published later, and one indicating that the steps for adding Gruntwork Pipelines is in the Gruntwork Pipelines deployment guide.

What do you think?

Reply:
> one indicating that the landing zone migration guide will be published later,

Can we just point to the existing Landing Zone guide? I'd hate to create a future work stream if it isn't necessary. I guess you're thinking that a migration guide would cover how to get from the legacy set up where some of the account-baseline functionality (like cloudtrail and cross-account-iam-roles) are in their own, dedicated modules. Or do I misunderstand what the landing zone migration guide would cover?

Reply:
Agreed on the proposed changes, including Ben's note!

Reply:
> I guess you're thinking that a migration guide would cover how to get from the legacy set up where some of the account-baseline functionality (like cloudtrail and cross-account-iam-roles) are in their own, dedicated modules. Or do I misunderstand what the landing zone migration guide would cover?

This is what I am thinking. A way to import your existing fragmented landing zone modules (`iam-groups`, `cross-account-iam-roles`, etc) into the `account-baseline-app` module. The landing zone deployment guide assumes a blank account, so it is useful for new accounts, but it won't be useful for customers who are migrating their existing account to landing zone (as the roles and stuff will conflict).

Hence, I don't think we should even point customers to the landing zone deployment guide because that would lead customers down the wrong path.

Reply:
My thought was that we'd announce the new ref arch in the blog, and link from that announcement to this migration guide. We can describe all the differences and promote it there, I feel.

Reply:
excuse my ignorance, but what is "Ref Arch 2.0" is there some additional step to take from "Ref Arch 1.0" -> "Service Catalog" -> "Ref Arch 2.0" ?

Reply:
Ref Arch 2.0 is the name we've been using for the refactored reference architecture that uses the service catalog and incorporates the CI/CD pipeline and the Landing Zone. It's automatically generated from the architecture catalog, which we haven't yet released to customers. It's depicted in our marketing copy here: https://gruntwork.io/reference-architecture/

Reply:
@bendavies Good question, and I didn't realize this could be a point of confusion. I'll (or whoever writes it will) point out in our announcement that the Service Catalog Ref Arch _is_ Ref Arch 2.0.